designs/2025-base-path-in-config-objects/README.md

```diff
@@ -0,0 +1,89 @@
+- Repo: eslint/rewrite (`@eslint/config-array`), eslint/eslint
+- Start Date: 2025-03-31
+- RFC PR: https://github.com/eslint/rfcs/pull/131
+- Authors: Milos Djermanovic
+
+# Support `basePath` property in config objects
+
+## Summary
+
+Currently, config array has a single property `basePath` (string) at the top level. When used from `eslint`, this property is set to the location of the config file, or the current working directory (when `--no-config-lookup` or `--config` options are used). `files` and `ignores` patterns in all config objects are treated as relative to the config array's `basePath`.
+
+This RFC proposes allowing config objects to specify their own `basePath` property. When present, config object's `basePath` overrides config array's `basePath`, meaning that `files` and `ignores` patterns in that config object should be treated as relative to the config object's `basePath` instead of the config array's `basePath`. Also, when `basePath` is present in a config object that has neither `files` nor `ignores`, then the config object applies to files in `basePath` and its subdirectories only.
```

Comment:
I've realized the behavior of `basePath` in configuration objects that have neither `files` nor `ignores` was not considered, so I updated this RFC in https://github.com/eslint/rfcs/pull/131/commits/09409edca7690fd5738c7105ef06b9d1c415d191, and the proof of concept accordingly in https://github.com/mdjermanovic/rewrite/pull/1/commits/d2ea0d79d44f2f6164b57d41698dcfb0a58e1fbe.